docs/metrics/bleu.md

```diff
@@ -0,0 +1,132 @@
+---
+search:
+  exclude: true
+---
+
+# BLEU (**B**i**L**ingual **E**valuation **U**nderstudy)
+
+!!! info inline end "BLEU vs. Precision"
+
+    BLEU can be thought of as an analog to [precision](precision.md) for text comparisons.
+
+The BLEU score is a metric commonly used in a variety of NLP tasks ranging from Machine Translation to Text Summarization, to evaluate the quality of candidate texts. It quantifies the similarity between the candidate text and reference text as a score between 0 and 1 — 0 meaning there is no overlap with the ground truth text, and 1 meaning there is a perfect overlap. As the name suggests, it was originally developed for evaluating machine-translation models, but has since been adapted to many different tasks within NLP due to its dynamic nature for measuring textual similarity.
+
+## Implementation Details
+### Definition
+The BLEU score is mathematically defined as:
+
+$$\begin{align*} \text{BLEU} &= \text{Brevity Penalty} \times \text{n-Gram Overlap} \\
+&= \min\left(1, \exp\left(1 - \frac{\text{reference length}}{\text{output length}}\right)\right) \times \left(\prod_{i=1}^{4}\text{i-Gram Precision}^{1/4}\right)
```

Comment:
I think adding a whitespace between 1 and exp(...) makes it look better. What do you think?

```
&= \min\left(1, \, \exp\left(1 - \frac{\text{reference length}}{\text{output length}}\right)\right) \times \left(\prod_{i=1}^{4}\text{i-Gram Precision}^{1/4}\right)
```

<img width="763" alt="image" src="https://github.com/kolenaIO/kolena/assets/17771952/84e1a32c-db31-485d-aad6-d5247ff8296b">